ITMO_FS/filters/univariate/measures.py

```diff
@@ -867,6 +868,185 @@ def modified_t_score(X, y):
+        [1 - __rbf_vectors(X[index], X[i], weights, p, radius) for i in range(X.shape[0]) if
+         i not in skip and i != also_skip])
+
+
+def __product_sequence_derivative(X, index, skip, weights, p, radius):
+    return np.sum(
+        [__rbf_derivative(index, i, weights, p, radius) * __prod_seq_func(X, index, skip, weights, p, radius, i) for i
+         in range(X.shape[0]) if
+         i not in skip],
+        axis=0)
+
+
+def __K_derivative(X, index, weights, p, radius):
+    sum_func = lambda skip: __rbf_derivative(X[index], X[skip], weights, p, radius) * \
+                            __prod_seq_func(X, index, [skip, index], weights, p, radius) + \
+                            __rbf_vectors(X[index], X[skip], weights, p, radius) * \
+                            __product_sequence_derivative(X, index, [index, skip], weights, p, radius)
+
+    return __product_sequence_derivative(X, index, [index], weights, p, radius) + np.sum(
+        [sum_func(i) for i in range(X.shape[0]) if i != index], axis=0)
+
+
+def __count_K_all(X, index, weights, p, radius):
+    all_distances = [__rbf(__weighted_minkowski_distance(X[index], X[t], weights, p), radius) for t in
+                     range(X.shape[0]) if t != index]
+    distances_minus = np.prod([1 - dist for dist in all_distances])
+    distances_without = [dist * distances_minus / (1 - dist) for dist in all_distances]
+    return distances_minus + sum(distances_without), distances_without, distances_minus
+
+
+def __single_mass_derivative(X, i, j, weights, p, radius):
+    K, _, distances_minus = __count_K_all(X, i, weights, p, radius)
+    return (K * __rbf_derivative(X[i], X[j], weights, p, radius) - __K_derivative(X, i, weights, p, radius) *
+            __rbf_vectors(X[i], X[j], weights, p, radius)) * distances_minus / (K ** 2) + \
+           __rbf_vectors(X[i], X[j], weights, p, radius) / \
+           K * __product_sequence_derivative(X, i, [i], weights, p, radius)
+
+
+def __mass_star_derivative(X, i, weights, p, radius):
+    K, _, distances_minus = __count_K_all(X, i, weights, p, radius)
+    return (K * __product_sequence_derivative(X, i, [i], weights, p, radius) -
+            __K_derivative(X, i, weights, p, radius) * distances_minus) / (K ** 2)
+
+
+def __y_derivative(X, i, weights, p, radius, y):
+    y_der = [0 for i in range(len(weights))]
+    y_lab = [y[j] for j in range(len(y)) if j != i]
+    for j in range(X.shape[0]):
+        if j == i:
+            continue
+        y_der += __single_mass_derivative(X, i, j, weights, p, radius) * y[j]
+    y_der += __mass_star_derivative(X, i, weights, p, radius) * (max(y_lab) + min(y_lab)) / 2
+    return y_der
+
+
+def __update_weights(X, y, alpha, weights, p, radius):
+    return weights + alpha * 2.0 / X.shape[0] * -1 * \
+           np.sum([__y_derivative(X, i, weights, p, radius, y) for i in range(X.shape[0])], axis=0)
+
+
+def weighted_evidential_regression(X, y, alpha=0.01, num_epochs=1000, p=2, k=None, radius=5.0):
```

Comment:
Does it a filter algorithm? 
Not embedded one?

Reply:
Yes it is more of an embedded method, but it fits better the pipeline of univariate filter, as it needs cutting rule and provides feature metric that can be used as a rank.